packages/libs/error-stack/src/fmt/mod.rs

```diff
@@ -32,57 +32,61 @@
 //! # #![cfg_attr(not(nightly), allow(dead_code, unused_variables, unused_imports))]
 //! use std::io::{Error, ErrorKind};
-//! use error_stack::{
-//!     fmt::{Emit},
-//!     Report
-//! };
+//! use error_stack::Report;
+//! use error_stack::fmt::Emit;
+//!
+//! struct ErrorCode(u64);
+//! struct Suggestion(&'static str);
+//! struct Warning(&'static str);
+//! struct Info(&'static str);
 //!
 //! // This hook will never be called, because a later invocation of `install_debug_hook` overwrites
 //! // the hook for the type `u64`.
-//! Report::install_debug_hook::<u64>(|_, _| Emit::next("will never be called"));
+//! Report::install_debug_hook::<ErrorCode>(|_, _| vec![Emit::next("will never be called")]);
 //!
 //! // `HookContext` always has a type parameter, which needs to be the same as the type of the
 //! // value, we use `HookContext` here as storage, to store values specific to this hook.
 //! // Here we make use of the auto-incrementing feature.
-//! // The incrementation is type specific, meaning that `ctx.increment()` for the `u32` hook will
-//! // not influence the counter of the `u64` or `u16` hook.
-//! Report::install_debug_hook::<u32>(|_, ctx| Emit::next(format!("u32 value {}", ctx.increment())));
+//! // The incrementation is type specific, meaning that `ctx.increment()` for the `Suggestion` hook
+//! // will not influence the counter of the `ErrorCode` or `Warning` hook.
+//! Report::install_debug_hook::<Suggestion>(|Suggestion(val), ctx| vec![Emit::next(format!("Suggestion {}: {val}", ctx.increment() + 1))]);
 //!
 //! // we do not need to make use of the context, to either store a value for the duration of the
 //! // rendering, or to render additional text, which is why we omit the parameter.
-//! Report::install_debug_hook::<u64>(|val, _| Emit::next(format!("u64 value ({val})")));
+//! Report::install_debug_hook::<ErrorCode>(|ErrorCode(val), _| vec![Emit::next(format!("Error ({val})"))]);
+//!
+//! Report::install_debug_hook::<Warning>(|Warning(val), ctx| {
+//!     let idx = ctx.increment() + 1;
 //!
-//! Report::install_debug_hook::<u16>(|_, ctx| {
 //!     // we set a value, which will be removed on non-alternate views
 //!     // and is going to be appended to the actual return value.
 //!     if ctx.alternate() {
-//!         ctx.attach_snippet("Look! I was rendered from a `u16`");
+//!         ctx.attach_snippet(format!("Warning {idx}:\n  {val}"));
 //!     }
-//!     Emit::next("For more information, look down below")
+//!
+//!     vec![Emit::next(format!("Warning ({idx}) occurred"))]
 //!  });
 //!
 //! // you can use arbitrary values as arguments, just make sure that you won't repeat them.
-//! // here we use [`Emit::defer`], this means that this value will be put at the end of the group.
-//! Report::install_debug_hook::<String>(|val, _| Emit::defer(val));
+//! // here we use [`Diagnostics::defer`], this means that this value will be put at the end of the group.
```

Comment:
incorrect docs, should be `Vec<Emit>`

Reply:
fixed in [4892398](https://github.com/hashintel/hash/pull/976/commits/4892398d338cb681fe7bbd5443b584fdc6383fad)